web/history.jsp

```diff
@@ -246,7 +246,7 @@ document.domReady.push(function() {domReadyHistory();});
                     if (entry.isActive()) {
                         String rp = uriEncodedName;
             %>
-            <td><a href="<%= context + Prefix.HIST_L + rp %>#<%= rev %>"
+            <td><a href="<%= request.getRequestURL().append('?').append(request.getQueryString()) %>#<%= rev %>"
```

Comment:
The `#` and `rev` can be appended as well. That said, the cascade of appends is a bit hard to read.